### ISSUE TITLE
v2.6.0: `Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.`

### ISSUE THREAD

**Issue by garyking** [typescript-eslint/typescript-eslint#1159]
**What code were you trying to parse?**

A new, empty file.

**What did you expect to happen?**

No `eslint` error should occur in the VS Code IDE for the new file.

**What actually happened?**

When creating a new file in my `src` dir (which is in `tsconfig.json`) in any of my TypeScript projects, in VS Code, the following error immediately appears on the file (via the VS Code ESLint extension):

```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: src/test.ts.
The file must be included in at least one of the projects provided.
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.6.0` |
| `TypeScript`                | `3.6.3` |
| `ESLint`                    | `6.6.0` |
| `node`                      | `12.12.0` |
| `npm`                       | `6.12.1` |

Also, MacOS 10.14.6 and VS Code 1.39.2.

I know this issue has been mentioned several times in this repo, but it seems like they're all closed. The issue was supposedly resolved in 2.6.0 of the parser, but it's not fixed for me. 

The new file I am creating are definitely in a directory that is included in `tsconfig.json`. The error disappears when I use `Reload Window` in VS Code, but I'd rather not want to do that for every new file I create, primarily because I have several scripts that run when a VS Code workspace is opened, and so I'd have to wait a few minutes after reloading the window.

.eslintrc.js:

```js
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "tsconfig.json"
  },
  "plugins": ["@typescript-eslint"],
  "extends": ["plugin:@typescript-eslint/recommended"]
}
```

tsconfig.json:

```json
{
  "compilerOptions": {
    "esModuleInterop": true,
    "module": "CommonJS",
    "outDir": "./dist/",
    "resolveJsonModule": true,
    "rootDir": "./src/",
    "strict": true,
    "target": "ES2020"
  },
  "include": ["./src/"],
  "exclude": [
    "**/build/**",
    "**/coverage/**",
    "**/dist/**",
    "**/node_modules/**"
  ]
}
```

**Comment by ccornici** [comment]
Same issue, had it on 2.5.0 and now updated to 2.6.0 and still doesn't work.

Fails both in VSCode with the ESLint extension and in WebStorm.

Restarting both "fixes" it.

Very annoying.

**Comment by ccornici** [comment]
```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: D:\Projects\Core\proj-name\test-integration\test-utils\generators\weww.ts.
The file must be included in at least one of the projects provided.
```

tsconf
```{
  "compilerOptions": {
    "esModuleInterop": true,
    "baseUrl": ".",
    "target": "es2018",
    "module": "commonjs",
    "moduleResolution": "node",
    "sourceMap": true,
    "outDir": "./built",
    "rootDir": "./",
    "strict": true,
    "noImplicitAny": false, //TODO put true after conversion is done
    "allowJs": true, //TODO delete after conversion is done
    "strictNullChecks": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "resolveJsonModule": true
  },
  "include": [
    "**/src/**/*.ts",
    "**/test/**/*.ts",
    "**/test-integration/**/*.ts"
  ]
}
```

eslintrc

**Comment by avinersan** [comment]
I am also encountering this issue.

Versions
- macOS 10.14.6
- eslint 6.6.0
- typescript 3.4.5
- @typescript-eslint/parser 2.6.0

**Comment by bradzacher** [comment]
I'm unable to repro this - it's working fine in the tests and locally within our repo.

Unfortunately there's no easy way for me to help here without github repos to test on, because debugging extensions isn't as simple as running the cli with a debug flag.

There's one thing I can ask you to do, but it's a bit involved.
1) open your project in your project in your editor
2) find your eslint install in your root folder, and open `lib/api.js` (i.e. `node_modules/eslint/lib/api.js`)
3) just below the comment header, add a new line: `process.env.DEBUG = 'typescript-eslint:*';`, and save the file.
4) restart your IDE.
5) open the extension output channel (in vscode you can do `command + shift + p` and pick the option "eslint: show output channel"
6) create a new file in your workspace, the same way you were doing to repro the error.
7) inspect the output logs - you should see a bunch of logs lines that are prefixed with like `2019-11-01T17:07:18.514Z typescript-eslint:`.
    - if you do not see lines matching the above, then your eslint extension isn't using the local install you modified, so you'll have look again to find the correct one.
8) copy+paste the output logs here.

Without this there's not much more I can do.

**Comment by Reeywhaar** [comment]
I made what @bradzacher been asking, here is output:

```
[Info  - 12:52:31 AM] ESLint server stopped.
[Info  - 12:52:31 AM] ESLint server running in node v10.11.0
[Info  - 12:52:31 AM] ESLint server is running.
[Info  - 12:52:34 AM] ESLint library loaded from: %project%/node_modules/eslint/lib/api.js
2019-11-01T21:53:16.958Z typescript-eslint:typescript-estree:createProjectProgram Creating project program for: %project%/src/path1/file1.tsx
2019-11-01T21:53:16.960Z typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. %project%/src/path1/file1.tsx
2019-11-01T21:53:16.960Z typescript-eslint:typescript-estree:createWatchProgram Creating watch program for %project%/tsconfig.json.
2019-11-01T21:53:25.698Z typescript-eslint:typescript-estree:createProjectProgram Creating project program for: %project%/src/path2/file2.tsx
2019-11-01T21:53:25.983Z typescript-eslint:typescript-estree:createWatchProgram Found existing program for file. %project%/src/path2/file2.tsx
// here I created the file
2019-11-01T21:53:36.256Z typescript-eslint:typescript-estree:createProjectProgram Creating project program for: %project%/src/path2/New.tsx
2019-11-01T21:53:36.256Z typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. %project%/src/path2/New.tsx
2019-11-01T21:53:36.258Z typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. %project%/src/path2/New.tsx
2019-11-01T21:53:39.927Z typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. %project%/src/path2/New.tsx
```

----

OSX 10.14.6
VSCode 1.39.2 with ESLint 1.9.1

eslintrc.js
```
module.exports = {
  parser: 'babel-eslint',
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:prettier/recommended',
  ],
  env: {
    browser: true,
    node: true,
    es6: true,
  },
  plugins: ['react', 'react-hooks', 'jsx-a11y', 'prettier'],
  parserOptions: {
    ecmaVersion: 6,
    sourceType: 'module',
    ecmaFeatures: {
      modules: true,
      jsx: true,
      experimentalDecorators: true,
      legacyDecorators: true,
    },
  },
  rules: {
    ...
  },
  overrides: [
    {
      files: ['*.ts', '*.tsx'],
      plugins: ['@typescript-eslint'],
      parser: '@typescript-eslint/parser',
      parserOptions: {
        project: './tsconfig.json',
        tsconfigRootDir: __dirname,
        experimentalDecorators: true,
      },
      rules: {
        ....
      },
    },
    ...
  ],
  globals: {
     ...
  },
}
```

tsconfig.json
```
{
  "compilerOptions": {
    "target": "es2017",
    "module": "esnext",
    "lib": ["es2017", "dom"],
    "incremental": true,
    "jsx": "preserve",
    "jsxFactory": "React.createElement",
    "strict": true,
    "esModuleInterop": true,
    "moduleResolution": "node",
    "alwaysStrict": true,
    "sourceMap": true,
    "noImplicitAny": false,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "allowSyntheticDefaultImports": true,
    "skipLibCheck": true,
    "allowJs": true,
    "baseUrl": "./src",
    "outDir": "./build",
    "paths": {
      "@app/*": ["./*"]
    }
  },
  "include": ["./src/**/*"],
  "exclude": ["node_modules", "public"]
}
```

**Comment by bradzacher** [comment]
the logs are as I expected, and looking at your config, there's nothing unexpected.

I have attempted to create a small repro using the exact config linked by @Reeywhaar, and it does not repro for me.

Same macos version, same vscode and extension version.
Same eslint version, newer versions of our package, but the changes wouldn't effect it. (you didn't list a typescript version, but I tried with both 3.6.x and 3.4.5, and both worked fine...)

**Comment by Reeywhaar** [comment]
@bradzacher sure the logs are expected? I found this line rather strange:

```
2019-11-01T21:53:39.927Z typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. %project%/src/path2/New.tsx
```

Also, what I noticed, is that if I add file right after vscode start there is no error. But if I add after some delay then problem appears.

**Comment by Reeywhaar** [comment]
Made test repo, but found that my personal problem was in fact gone in 2.6.0, but present in 2.5.0

https://github.com/Reeywhaar/eslint-1159

**Comment by bradzacher** [comment]
> sure the logs are expected

Sorry I meant that they're the logs I expected to see in this error case.
There is only one code path to get to this hard fail case, and that is it (look in cache -> make sure all tsconfigs have programs -> trigger a folder update -> check for file deletions -> still nothing? return nothing)

Without a solid repro against 2.6.0 (or 2.7.0 which will release today), I'm going to assume this is fixed.

Feel free to comment if you can create a repro case against the latest version!

**Comment by kitety** [comment]
The same problem happened

**Comment by andywampir** [comment]
I have the same problem too.

**Comment by coolgk** [comment]
I have similar issue too

![image](https://user-images.githubusercontent.com/4009549/68548440-5922f700-0406-11ea-8eb3-b4af311fc600.png)

**Comment by bradzacher** [comment]
Please do not comment "I have the same problem" or similar.
It's just spam, and creates noise for everyone in the thread.
Use the +1 button on the issue.

If you have a repro case, feel free to comment then.

**Comment by lygstate** [comment]
My full eslint log with the same error
```
PS E:\CI-Cor-Ready\xemu\xemu-nm\server> npm run lint

> xemu-nm@0.0.1 lint E:\CI-Cor-Ready\xemu\xemu-nm\server
> eslint .

  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\tcp-client.js +0ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\tcp-client.js +0ms
  typescript-eslint:typescript-estree:createWatchProgram Creating watch program for e:\ci-cor-ready\xemu\xemu-nm\server\tsconfig.json. +1ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\tcp-server.js +3s
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\tcp-server.js +3s
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\tcp-server.js +1ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\tcp-server.js +21ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\ws-client-test.js +30ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\ws-client-test.js +7ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\ws-client-test.js +1ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\ws-client-test.js +22ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\ws-client.js +32ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\ws-client.js +8ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\ws-client.js +53ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\scripts\ws-client.js +27ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\app.controller.js +113ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.controller.js +33ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.controller.js +2ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.controller.js +23ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\app.module.js +62ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.module.js +37ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.module.js +2ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.module.js +20ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\app.service.js +56ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.service.js +35ms
+4ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\app.service.js +25ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\main.js +61ms
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\main.js +31ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\main.js +1ms   
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\main.js +22ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\terminals\terminals.adapter.js +54ms     
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\terminals\terminals.adapter.js +32ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\terminals\terminals.adapter.js +1ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\terminals\terminals.adapter.js +9ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\terminals\terminals.gateway.js +122ms    
  typescript-eslint:typescript-estree:createWatchProgram File did not belong to any existing programs, moving to create/update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\terminals\terminals.gateway.js +153ms
  typescript-eslint:typescript-estree:createWatchProgram File was not found in program - triggering folder update. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\terminals\terminals.gateway.js +2ms
  typescript-eslint:typescript-estree:createWatchProgram File was still not found in program after directory update - checking file deletions. e:\ci-cor-ready\xemu\xemu-nm\server\dist\src\terminals\terminals.gateway.js +7ms
  typescript-eslint:typescript-estree:createProjectProgram Creating project program for: E:\CI-Cor-Ready\xemu\xemu-nm\server\scripts\ws-tcp-relay.js +392ms
  typescript-eslint:typescript-estree:createWatchProgram Found existing program for file. e:\ci-cor-ready\xemu\xemu-nm\server\scripts\ws-tcp-relay.js +372ms
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\tcp-client.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\scripts\tcp-client.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\tcp-server.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\scripts\tcp-server.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\ws-client-test.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\scripts\ws-client-test.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\ws-client.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\scripts\ws-client.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\app.controller.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\src\app.controller.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\app.module.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\src\app.module.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\app.service.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\src\app.service.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\main.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\src\main.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\terminals\terminals.adapter.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\src\terminals\terminals.adapter.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\src\terminals\terminals.gateway.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: dist\src\terminals\terminals.gateway.js.
The file must be included in at least one of the projects provided

E:\CI-Cor-Ready\xemu\xemu-nm\server\scripts\ws-tcp-relay.js
  20:7  warning  Unexpected console statement  no-console
  28:7  warning  Unexpected console statement  no-console
  33:5  warning  Unexpected console statement  no-console
  38:3  warning  Unexpected console statement  no-console

✖ 14 problems (10 errors, 4 warnings)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! xemu-nm@0.0.1 lint: `eslint .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the xemu-nm@0.0.1 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\lygstate\AppData\Roaming\npm-cache\_logs\2019-11-13T17_40_57_681Z-debug.log
```

tsconfig.json
```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es2017",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "checkJs": true,
    "paths": {
      "*" : ["./@types/*", "./node_modules/@types/*"]
    }
  },
  "exclude": ["node_modules", "dist"]
}
```

**Comment by bradzacher** [comment]
@lygstate 

```
E:\CI-Cor-Ready\xemu\xemu-nm\server\dist\scripts\ws-client-test.js
                                    ^^^^^
```

You should add `dist` to your `.eslintignore`.
The errors you're getting are expected, because `dist` is not included in the `tsconfig.json` you pass in.

**Comment by garyking** [comment]
@bradzacher I managed to narrow down the issue as far as I could. First, here are my files:

```tsconfig.json```:

```json
{
  "compilerOptions": {
    "module": "CommonJS",
    "outDir": "./dist/",
    "rootDir": "./src/",
    "strict": true
  },
  "include": ["./src/"]
}
```

```.eslintrc.js```:

```js
module.exports = {
  extends: ['plugin:@typescript-eslint/recommended'],
  parserOptions: {
    project: './tsconfig.json',
  },
};
```

I'm using the VS Code ESLint extension.

In VS Code, I create a file, like `src/a.ts`. I get the error message. If I `Reload Window`, then the error message is gone. Just like the issue has been described, so far.

The key is `"include": ["./src/"]` in `tsconfig.json`. If you change it to `"include": ["./src"]` (so, the final slash is removed), then the error will not appear.

**Comment by bradzacher** [comment]
woo hoo, awesome @garyking - you've found the repro! Thanks

Typescript doesn't like to be consistent, apparently 😞 
If you provide `./src/`, then it will ask us to watch the path `<absolute_path>/src/`.
If you provide `./src`, then it will ask us to watch the path `<absolute_path>/src`.

The code right now assumes the latter is how the watch path will always look.
So when you create a the file: `./src/foo/bar/baz.ts`, this is essentially what happens:
- look for watch callback for path `<absolute_path>/src/foo/bar`
- look for watch callback for path `<absolute_path>/src/foo`
- look for watch callback for path `<absolute_path>/src`
    - typescript did register on path `<absolute_path>/src/`, but string `===` match means it doesn't match, ofc. 
- no watch callback found for folder
- assume file must not be within tsconfig, and error

Now that I know the problem it's easy to normalise the paths and fix this.

**Comment by diegohaz** [comment]
Another reproduction:
1. Clone https://github.com/reakit/reakit/ and open it with VSCode
2. Rename any `.ts`/`.tsx` file on VSCode
3. See the error
    ![image](https://user-images.githubusercontent.com/3068563/68861291-8e239800-06eb-11ea-87f6-78f76e80f9fc.png)

**Comment by bradzacher** [comment]
@diegohaz - your problem is logged in https://github.com/typescript-eslint/typescript-eslint/issues/1110
Workaround is to add an `include` to your tsconfig.

**Comment by shellscape** [comment]
I'm still seeing the reported issue on 2.9.0 with the following `include`s:

```json
  "include": [
    "./**/*.{js,ts}"
  ]
```

and

```json
  "include": [
    "./**/*.*"
  ]
```

**Comment by bradzacher** [comment]
please don't comment on old issues.
open a new one with as much detail as you can provide.
